apps/fluent-tester/src/E2E/Switch/specs/Switch.spec.win.ts

```diff
@@ -0,0 +1,65 @@
+import NavigateAppPage from '../../common/NavigateAppPage.win';
+import SwitchPageObject from '../pages/SwitchPageObject';
+import { PAGE_TIMEOUT, BOOT_APP_TIMEOUT, BUTTON_A11Y_ROLE } from '../../common/consts';
+
+// Before testing begins, allow up to 60 seconds for app to open
+describe('Switch Testing Initialization', function () {
+  it('Wait for app load', () => {
+    NavigateAppPage.waitForPageDisplayed(BOOT_APP_TIMEOUT);
+    expect(NavigateAppPage.isPageLoaded()).toBeTruthy(NavigateAppPage.ERRORMESSAGE_APPLOAD);
+  });
+
+  it('Click and navigate to Switch test page', () => {
+    /* Scroll to component test page button in scrollview if not already visible*/
+    SwitchPageObject.scrollToComponentButton();
+    SwitchPageObject.waitForButtonDisplayed(PAGE_TIMEOUT);
+
+    /* Click on component button to navigate to test page */
+    NavigateAppPage.clickAndGoToSwitchPage();
+    SwitchPageObject.waitForPageDisplayed(PAGE_TIMEOUT);
+
+    expect(SwitchPageObject.isPageLoaded()).toBeTruthy(SwitchPageObject.ERRORMESSAGE_PAGELOAD);
+    expect(SwitchPageObject.didAssertPopup()).toBeFalsy(SwitchPageObject.ERRORMESSAGE_ASSERT); // Ensure no asserts popped up
+  });
+});
+
+describe('Switch Accessibility Testing', () => {
+  /* Scrolls and waits for the Switch to be visible on the Test Page */
+  beforeEach(() => {
+    SwitchPageObject.scrollToTestElement();
+    SwitchPageObject.waitForPrimaryElementDisplayed(PAGE_TIMEOUT);
+  });
+
+  it('Switch - Validate accessibilityRole is correct', () => {
+    expect(SwitchPageObject.getAccessibilityRole()).toEqual(BUTTON_A11Y_ROLE);
+    expect(SwitchPageObject.didAssertPopup()).toBeFalsy(SwitchPageObject.ERRORMESSAGE_ASSERT);
+  });
+});
+
+describe('Switch Functional Testing', () => {
```

Comment:
What about keyboard interaction? Does the switch toggle on space/enter? That's easily added here too (see Checkbox again)

Reply:
I had that in a accessibility branch but that should have been included in this one. Thanks for catching that!